### DISCUSSION TITLE
Yolov8 live inferencing without using Ultralytics Library.

### DISCUSSION THREAD

**Discussion by Ammar-hala** [ultralytics/ultralytics#7133] [Q&A]
Can we do live inferencing on yolov8 models without using ultralytics library? I want to setup the codebase just for live inferencing on models but don't want to use the library directly.

**Comment by pderrenger** [comment]
@Ammar-hala yes, you can perform live inferencing with YOLOv8 models without directly using the Ultralytics library. To do this, you would typically export the YOLOv8 model to a format compatible with your preferred inference framework, such as ONNX, TensorRT, or TorchScript. Once exported, you can use the appropriate tools and libraries for that format to perform live inferencing.

Here's a general outline of the steps you might follow:

1. Export the YOLOv8 model to a deployment-ready format using the `export` mode. For example, you can export to ONNX format with dynamic axes to accommodate varying image sizes.

2. Load the exported model into your chosen inference framework. For example, if you exported to ONNX, you could use the ONNX Runtime for inference.

3. Set up your live video feed or camera stream to capture frames in real-time.

4. Preprocess the input frames as required by the model (e.g., resizing, normalization).

5. Perform inference on each frame using the loaded model.

6. Post-process the model outputs to extract the desired information, such as bounding boxes, classes, and confidence scores.

7. Display the results on the live video feed or take appropriate actions based on the detections.

For detailed instructions on exporting models and using them in different frameworks, please refer to the Export section of the Ultralytics Docs. Keep in mind that while you can use the model without the Ultralytics library, you will need to handle all aspects of preprocessing, inference, and postprocessing manually.
